docker-compose.yaml

```diff
@@ -72,4 +73,26 @@ services:
       - SAMBA_DC_ADMIN_PASSWD=SuperSecretPassw0rd
       - SAMBA_DC_DNS_BACKEND=SAMBA_INTERNAL
 
+  redis:
+    image: redis:7.0.11
+    ports:
+      - '6379:6379'
+    command: >
+      --requirepass password
+
+  couchbase:
+    image: couchbase:7.2.0
+    ports:
+      - '8091-8097:8091-8097'
+      - '9123:9123'
+      - '11207:11207'
+      - '11210:11210'
+      - '11280:11280'
+      - '18097:18097'
 
+  cassandra:
+    image: cassandra:4.0
+    ports:
+      - "9042:9042"
+    volumes:
+      - ./cassandra.yaml:/etc/cassandra/cassandra.yaml
```

Comment:
nit: could we move the cassandra.yaml file out of the top-level directory? Since it is only for testing, I find it to be out of place here. Maybe we could put it in `./testdata`?